docs/reference/module-api/module-overview.md

```diff
@@ -908,12 +826,146 @@ module.exports = {
+
+Note that it can often be simpler to add an event handler or `await` a method in an API route instead.
+
+`ours` could also be an object with a `before` property and a `middleware` property, in which case it would run `before` the middleware of the module with the specified name
+
+```javascript
+// modules/limiter/index.js
+module.exports = {
+  // ...
+  middleware(self, options) {
+    return {
+      ours(req, res, next) {
+        // Restrict access by IP address, in a crude way.
+        const allowlist = [ '127.0.0.1', '::1' ];
+
+        if (!allowlist.includes(req.connection.remoteAddress)) {
+          return res.status(403).send('forbidden');
+        }
+        return next();
+      }
+    };
+  }
+};
+```
+
+If the middleware function must run before another named middleware function, set the returned object key to an object with `before` and `middleware` properties. `before` would be set to the name of a module whose middleware must run after the new function, which is now the value of `middleware`.
+
+```javascript
+// modules/limiter/index.js
+module.exports = {
+  // ...
+  middleware(self, options) {
+    return {
+      ours: {
+        // 👇 Same as above, but with `before` and `middleware` properties.
+        before: '@apostrophecms/login',
+        middleware: function (req, res, next) {
+          // Restrict access by IP address, in a crude way.
+          const allowlist = [ '127.0.0.1', '::1' ];
+
+          if (!allowlist.includes(req.connection.remoteAddress)) {
+            return res.status(403).send('forbidden');
+          }
+          return next();
+        }
+      }
+    };
+  }
+};
+```
+
 ### `tasks(self)`
 
-## "Cascading" settings
+`tasks` takes the module as an argument and returns an object of functions that define command-line tasks. Task properties include:
+
+
+| Property | Description |
+| ------- | ------- |
+| `usage` | A string describing the task and how to use it. It is printed on the command-line. |
+| `task` | The task function. It maybe asynchronous. |
+| `afterModuleInit` | Set to `true` to run the task after modules are initiated but *before* they are fully "awake".
+| `exitAfter` | Only relevant if `afterModuleInit` is `true`. Set to `false` to *avoid* exiting the Apostrophe process on completion. Uncommon. |
+
+Task functions takes the object `argv` as an argument, which includes the arguments passed after the task command. As documented by the [Boring](https://www.npmjs.com/package/boring) utility:
+
+> When someone types:
+>
+> ```
+> node app jump sideways --foo --bar=whee --super-cool=totally
+> ```
+>
+> You get:
+>
+> ```
+> {
+>   _: [ "jump", "sideways"],
+>   foo: true,
+>   bar: "whee",
+>   "super-cool": "totally"
+> }
+> ```
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  tasks(self, options) {
+    return {
+      // Since the module is named product, then you can run this CLI task by
```

Comment:
```suggestion
      // Since the module is named product, then you can run this command line task by
```
Just trying to be consistent where it makes sense. If we use CLI more, this is fine.